### PR TITLE
check first instanceof Blob

### PR DESCRIPTION
**Pull request by spirinvladimir**
This changes are only for IndexedDB.

Main goal of PR is check type of value before check browser's blob support.

**Before changes**:
First check browser's blob support. Then if needed check value type.
**After changes**:
First check value type. Then if needed check browser's blob support.

Browser's blob support check make fake write blob to IndexedDB. And store result of check in private variable [supportsBlobs](https://github.com/mozilla/localForage/blob/master/src/drivers/indexeddb.js#L18)
Second browser's blob support check will use ```supportsBlobs``` without fake write blob to IndexedDB. 

I add 3 new tests:
* Save non blob value should not trigger browser's blob support check with fake save blob.
* Save blob value should trigger browser's blob support check with fake save blob.
* Second save blob value should use variable ```supportsBlobs``` and do not check fake save blob.

Blob support check functions and variable ```supportsBlobs``` are private. That is why I use mock db.transaction. After tests I return db.transaction to original state.

Issue is here:
https://github.com/mozilla/localForage/issues/476

### CODE REVIEW

File: dist/localforage.nopromises.js [hunk 1]
```diff
@@ -718,7 +718,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	            dbInfo.db = dbContext.db = db;
 	            self._dbInfo = dbInfo;
 	            // Share the final connection amongst related localForages.
-	            for (var k in forages) {
+	            for (var k = 0; k < forages.length; k++) {
```
Comment (tofumatt) [inline dist/localforage.nopromises.js]: Interesting that this change was made by the compiler.

**Comment by tofumatt** [comment]
Looks good; can you add a test for this?

File: dist/localforage.nopromises.js (hunk 1 above)
Comment (sebweaver) [inline dist/localforage.nopromises.js]: It's related to https://github.com/mozilla/localForage/commit/47aaaa82c7c86416e9e000973c5b105ca868e8cd

Comment (tofumatt) [inline dist/localforage.nopromises.js]: Oh right, forgot to add the dust files into that commit, oops!

**Comment by spirinvladimir** [comment]
I'v added 3 tests.

File: test/test.api.js [hunk 2]
```diff
@@ -223,6 +223,68 @@ DRIVERS.forEach(function(driverName) {
             });
         });
 
+        if (driverName === localforage.INDEXEDDB) {
```
Comment (tofumatt) [inline test/test.api.js]: I don't see any `expect` calls in these tests so I don't see what they're testing for. If you want `called` to be more than one you'll want to test for its value with `expect`, which I think is what you're going for.

Comment (tofumatt) [inline test/test.api.js]: Thanks for adding the `expect` calls, but I'm curious: what _exactly_ is this testing for? I'm a bit confused and the PR doesn't have a description nor is this related to an issue.

What bug does this prevent and what are these tests doing?

**Comment by spirinvladimir** [comment]
Sorry for PR without description. I've add it now.

**Comment by tofumatt** [comment]
Aha, cool!

That makes sense, awesome. I'll review this quickly in the evening then, but it looks good. Should be able to merge it tonight, sorry for the back-and-forth! :smile: 

Thanks! :smile:

**Pull request merged**